data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/DataStreamDetector.java

```diff
@@ -0,0 +1,66 @@
+/*
```

Comment:
We have a new license header for new files.

https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md#license-headers

Please use this for all new files.